lib/internal/bootstrap/pre_execution.js

```diff
@@ -63,6 +63,14 @@ function initializeESMLoader() {
   }
 }
 
+function initializeFrozenIntrinsics() {
+  if (getOptionValue('--frozen-intrinsics')) {
+    process.emitWarning('The --frozen-intrinsics flag is experimental',
+                        'ExperimentalWarning');
+    require('internal/freeze_intrinsics');
```

Comment:
I would suggest exporting a function from `internal/freeze_intrinsics` that does the actual freezing, that might make the code a bit more flexible in the future.

Reply:
Done.